`@strapi/plugin-sdk` `package.json` validation too aggressive, cannot add additional exports

<!--
Hello 👋 Thank you for submitting an issue.

Before you start, please make sure your issue is understandable and reproducible.
To make your issue readable make sure you use valid Markdown syntax.

https://guides.github.com/features/mastering-markdown/

Please ensure you have also read and understand the contributing guide.

https://github.com/strapi/strapi/blob/main/CONTRIBUTING.md#reporting-an-issue
-->

## Bug report

### Required System information

<!-- Please ensure you are using the Node LTS version (18, 20, or 22) -->
<!-- Strapi v3 is no longer supported, please update to Strapi v4 -->
<!-- If you are reporting a frontend bug please provide error logs after setting STRAPI_ENFORCE_SOURCEMAPS=true in your .env -->
<!-- This environment variable makes frontend errors easier to read and trace -->

- Node.js version: 22.12.0
- NPM version: 10.9.0
- Strapi version: 5.6.0
- Database: MySQL
- Operating system: Windows 11
- Is your project Javascript or Typescript: Typescript

### Describe the bug

When updating my plugins, I want to also export from the package a collection of types that can be used by developers/users of the plugin (things like the shape of the Config, shape of endpoint data, etc).

Now that Strapi v5 has switched to using the `exports` map I wanted to add my types to this export map. 

```json
    "exports": {
        "./package.json": "./package.json",
        "./strapi-admin": {
            "types": "./dist/admin/src/index.d.ts",
            "source": "./admin/src/index.ts",
            "import": "./dist/admin/index.mjs",
            "require": "./dist/admin/index.js",
            "default": "./dist/admin/index.js"
        },
        "./strapi-server": {
            "types": "./dist/server/src/index.d.ts",
            "source": "./server/src/index.ts",
            "import": "./dist/server/index.mjs",
            "require": "./dist/server/index.js",
            "default": "./dist/server/index.js"
        },
        "./types/*": "./dist-types/*.d.ts" // <--- Addition
    }
``` 

After adding it, trying to `build`/`watch` with the `@strapi/plugin-sdk` (`strapi-plugin build`) I get the following error

```bash
[ERROR] There seems to be an unexpected error, try again with --debug for more information 

 ┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────┐│                                                                                                             
││   ValidationError: exports["./types/*"] must match the following: "/^\.\/.*\.json$/"                        
││   at createError (C:\Projects\GESDev\ges-v2\packages\editorjs\strapi-plugin-react-editorjs\node_modules\y   ││   up\lib\util\createValidation.js:54:21)                                                                    
││   at C:\Projects\GESDev\ges-v2\packages\editorjs\strapi-plugin-react-editorjs\node_modules\yup\lib\util\c   ││   reateValidation.js:72:107                                                                                 
││                                                                                                             
│└─────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```

I also tried to use the object form of an export entry, and got a similar error, indicating that it had to use the same format as the `strapi-admin`/`strapi-server` objects, which doesn't work in this case.

--- a/.changeset/smart-eyes-cross.md
+++ b/.changeset/smart-eyes-cross.md
@@ -1,0 +1,5 @@
+---
+"@strapi/sdk-plugin": patch
+---
+
+fix: allow any file extension in exports string

--- a/src/cli/commands/utils/pkg.ts
+++ b/src/cli/commands/utils/pkg.ts
@@ -34,10 +34,7 @@ const packageJsonSchema = yup.object({
                   })
                   .noUnknown(true);
               } else {
-                acc[key] = yup
-                  .string()
-                  .matches(/^\.\/.*\.json$/)
-                  .required();
+                acc[key] = yup.string().required();
               }
 
               return acc;

